### PR TITLE
atuin: 18.1.0 -> 18.2.0

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Changelog: https://github.com/atuinsh/atuin/releases/tag/v18.2.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
The hash is probably different because of some special casing of a file.

**Comment by msanft** [comment]
@ofborg build atuin

**Comment by VuiMuich** [comment]
Since this release fixes some annoying breaking change with nu shell I put this overlay in my flake setup until this is is live and it works great.
Just in case anybody else wants to use this, here you go:
```nix
    atuin = prev.atuin.overrideAttrs (oldAttrs: rec {
      version = "18.2.0";
      
      src = prev.fetchFromGitHub {
        owner = "atuinsh";
        repo = "atuin";
        rev = "v${version}";
        hash = "sha256-TTQ2XLqng7TMLnRsLDb/50yyHYuMSPZJ4H+7CEFWQQ0=";
      };

      cargoDeps = oldAttrs.cargoDeps.overrideAttrs (prev.lib.const {
        name = "atuin-${version}-vendor.tar.gz";
        inherit src;
        outputHash = "sha256-KMH19Op7uyb3Z/cjT6bdmO+JEp1o2n6rWRNYmn1+0hE=";
      });

      checkFlags = [
        # tries to make a network access
        "--skip=registration"
        # No such file or directory (os error 2)
        "--skip=sync"
        # PermissionDenied (Operation not permitted)
        "--skip=change_password"
        "--skip=multi_user_test"
        # Tries to touch files
        "--skip=build_aliases"
      ];
    });
```

File: pkgs/by-name/at/atuin/package.nix [hunk 1]
```diff
@@ -10,20 +10,16 @@
...
-    hash = "sha256-ddj8vHFTRBzeueSvY9kS1ZIcAID8k3MXrQkUVt04rQg=";
+    hash = "sha256-TTQ2XLqng7TMLnRsLDb/50yyHYuMSPZJ4H+7CEFWQQ0=";
   };
 
-  # TODO: unify this to one hash because updater do not support this
-  cargoHash =
-    if stdenv.isLinux
-    then "sha256-LKHBXm9ZThX96JjxJb8d7cRdhWL1t/3aG3Qq1TYBC74="
-    else "sha256-RSkC062XB5zy3lmI0OQhJfJ6FqFWXhpMPNIIqbrrlso=";
+  cargoHash = "sha256-KMH19Op7uyb3Z/cjT6bdmO+JEp1o2n6rWRNYmn1+0hE=";
```
Comment (msanft) [inline pkgs/by-name/at/atuin/package.nix:22]: darwin hash is ´sha256-mBOyo6bKipMfmsowQujeUpog12jXAiqx5CtkwCxquRU=´

**Review by r-vdp — APPROVED** [review]
Working well for me!

**Pull request merged**